File: min_rotated.java
Time Complexity : O(logn)

class Solution {
    public int findMin(int[] nums) {
        
        
        int high = nums.length -1 ;
        int low = 0;
        
        
        if ( nums.length == 1) { return nums[0];}
        
        if ( nums[low] < nums[high])
        {
            return nums[0];
        }
        
        
        
        while ( low <= high )
        {
            int mid = low + ( high -low) /2;
            
            
             if (nums[mid] > nums [mid+1] )
            {
                return nums[mid+1];
            }
            
             else if (  nums[mid] < nums[mid-1] )
            {
                return nums[mid];
            }
            
            
            else if ( nums[mid] > nums[0])
            {
                low = mid +1;
            }
            
            else 
            {
                high = mid-1;
            }
            
           
            
        }
        
        return -1;
    }
}

File: minrotated.py
# Time Complexity : O(log*n)
# Space Complexity : O(1)
# Did this code successfully run on Leetcode : Yes

class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = nums[0]
        left, right = 0, len(nums) -1
        while left<=right:
            if nums[left] < nums[right]:
                result = min(result, nums[left])
                break
            mid = (left + right) // 2
            result = min(result, nums[mid])
            if nums[mid] >= nums[left]:
                left = mid +1
            else:
                right = mid - 1
        return result
            
        
         
        